File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun rightSideView(root: TreeNode?): List<Int> {
        val res = mutableListOf<Int>()
        if(root == null)
            return res
        var root = root
        val que = ArrayDeque<TreeNode>()
        que.add(root)
        var temp = 0
        while(!que.isEmpty()){
            for (i in 0 .. que.size -1)
            {
                val root = que.removeFirst()
                temp = root.`val`
                if(root.left != null) que.add(root.left)
                if(root.right != null) que.add(root.right)
            }
            res.add(temp)
        }
        return res
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0
        def helper(root, maxv):
            if not root:
                return
            if root.val >= maxv:
                self.res += 1
            
            helper(root.left, max(maxv, root.val))
            helper(root.right, max(maxv, root.val))
            return root
        helper(root, float('-inf'))
        return self.res